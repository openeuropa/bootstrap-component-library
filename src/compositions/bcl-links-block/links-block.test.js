import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-links-block/data/data";

const template = "@oe-bcl/bcl-links-block/links-block.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Links block", () => {
  test("renders correctly", () => {
    expect.assertions(1);
    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test("renders correctly with horizontal", () => {
    expect.assertions(1);
    return expect(
      render({ ...demoData, direction: "horizontal" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with transparent background", () => {
    expect.assertions(1);
    return expect(
      render({ ...demoData, variant: "transparent" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with title", () => {
    expect.assertions(1);
    return expect(
      render({ ...demoData, title: "Links block test title", title_tag: "h6" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with horizontal and with title", () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoData,
        direction: "horizontal",
        title: "Links block test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with transparent background and with title", () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoData,
        variant: "transparent",
        title: "Links block test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });
});
