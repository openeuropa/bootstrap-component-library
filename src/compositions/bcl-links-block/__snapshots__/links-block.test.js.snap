// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Links block renders correctly 1`] = `
<jest>
  <div
    class="bg-light px-4 py-3"
  >
    <h2
      class="fw-bold pb-3 mb-3 border-bottom"
    >
      More information
    </h2>
    <ul
      class="ps-0 mb-0"
    >
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          European Commission
        </a>
      </li>
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Priorities
        </a>
      </li>
      <li
        class="list-unstyled"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Jobs, Growth and Investment
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - Links block renders correctly with horizontal 1`] = `
<jest>
  <div
    class="bg-light px-4 py-3"
  >
    <h2
      class="fw-bold pb-3 mb-3 border-bottom"
    >
      More information
    </h2>
    <ul
      class="ps-0 mb-0"
    >
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          European Commission
        </a>
      </li>
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Priorities
        </a>
      </li>
      <li
        class="list-unstyled"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Jobs, Growth and Investment
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - Links block renders correctly with horizontal and with title 1`] = `
<jest>
  <div
    class="bg-light px-4 py-3"
  >
    <h6
      class="fw-bold pb-3 mb-3 border-bottom"
    >
      Links block test title
    </h6>
    <ul
      class="ps-0 mb-0"
    >
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          European Commission
        </a>
      </li>
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Priorities
        </a>
      </li>
      <li
        class="list-unstyled"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Jobs, Growth and Investment
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - Links block renders correctly with title 1`] = `
<jest>
  <div
    class="bg-light px-4 py-3"
  >
    <h6
      class="fw-bold pb-3 mb-3 border-bottom"
    >
      Links block test title
    </h6>
    <ul
      class="ps-0 mb-0"
    >
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          European Commission
        </a>
      </li>
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Priorities
        </a>
      </li>
      <li
        class="list-unstyled"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Jobs, Growth and Investment
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - Links block renders correctly with transparent background 1`] = `
<jest>
  <div
    class="bg-light px-4 py-3"
  >
    <h2
      class="fw-bold pb-3 mb-3 border-bottom"
    >
      More information
    </h2>
    <ul
      class="ps-0 mb-0"
    >
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          European Commission
        </a>
      </li>
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Priorities
        </a>
      </li>
      <li
        class="list-unstyled"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Jobs, Growth and Investment
        </a>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - Links block renders correctly with transparent background and with title 1`] = `
<jest>
  <div
    class="bg-light px-4 py-3"
  >
    <h6
      class="fw-bold pb-3 mb-3 border-bottom"
    >
      Links block test title
    </h6>
    <ul
      class="ps-0 mb-0"
    >
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          European Commission
        </a>
      </li>
      <li
        class="list-unstyled me-4-5"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Priorities
        </a>
      </li>
      <li
        class="list-unstyled"
      >
        <a
          class="pb-3 d-inline-block"
          href="/example.html"
        >
          Jobs, Growth and Investment
        </a>
      </li>
    </ul>
  </div>
</jest>
`;
